lib/base/error-handler.js

```diff
@@ -4,122 +4,241 @@
+
+            "USE_WITH_OPTIONS" : "Use together with options",
+            "USE_PKGID_PKGINFO" : "packageId must be provided by using either the '--pkgid' or the '--pkginfofile' option",
+            "USE_GDB" : "Please use GDB to debug a native serivce",
+            "NOT_IMPLEMENTED" : "Not implemented",
+            "NOT_HANDLE_FILE" : "Don't know how to handle",
+            "NOT_OVERWRITE_DIR" : "Cannot overwrite the directory",
+
+            "UNMATCHED_DEVICE" : "No matched device",
+            "CANNOT_REMOVE_DEVICE" : "Cannot remove the device",
+            "UNMATCHED_DISPLAY_AFFINITY" : "Please use the same value for \"display\" and \"displayAffinity\"",
+
+            "FAILED_CALL_LUNA" : "luna-send command failed",
+            "FAILED_GET_PORT" : "Failed to get Debug port",
+            "FAILED_GET_SVCPATH" : "Failed to get service installation path",
+            "FAILED_GET_SSHKEY" : "Failed to get ssh private key",
+            "FAILED_TRANSMIT_FILE" : "File transmission error. Please try again",
+            "FAILED_CREATE_DIR" : "Cannot create directory in the destination path - permission denied",
+            "FAILED_REMOVE_PACKAGE" : "No packages installed or removed",
+            "FAILED_FIND_SERVICE" : "Failed to find a service with the specified name",
+
+            "CONNECTED_MULTI_DEVICE" : "Multiple devices are connected by novacom. Please specify the target name",
+            "SET_DEFAULT_MULTI_DEVICE" : "Multiple devices are set to default target. Please reset device list",
+            "NEED_ROOT_PERMISSION" : "Unable to connect to the target device. root access required",
+            "MISSING_CALLBACK" : "Missing completion callback",
+            "USING_WEBINSPECTOR" : "Web Inspector is already connected with another browser. Please close the previous connection",
+            "UNKNOWN_OPERATOR" : "Unknown operator",
+            "UNABLE_USE_SFTP" : "Unable to use sftp",
+        };
+
+    errMsgHdlr.finish = function(err, value) {
+        if(err) {
+            if (typeof(err) === "string") {
+                log.error(err.toString());
+                log.verbose(err.stack);
+            } else if (typeof(err) == "object") {
+                if (err.length === undefined) { // single error
+                    log.error(err.heading, err.message);
+                    log.verbose(err.stack);
+                } else if (err.length > 0) { // [service/system] + [tips] error
+                    for(const index in err) {
+                        log.error(err[index].heading, err[index].message);
+                    }
+                    log.verbose(err[0].stack);
+                }
+            }
+            cliControl.end(-1);
+        } else {
+            log.info('finish():', value);
+            if (value && value.msg) {
+                console.log(value.msg);
+            }
+            cliControl.end();
+        }
+    };
+
+    errMsgHdlr.getServiceErrMsg = function(service, errKey, errText) {
```

Comment:
The best thing I think is to call one AAA function and process each of getSyscallErrMsg, getService ErrMsg or getErrMsg from inside.

Let's think together if we can do this.

Reply:
Okay :)

Reply:
I integrated getErrMsg function();